eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/LightningMessageCodecs.scala

```diff
@@ -130,6 +168,66 @@ object LightningMessageCodecs {
       ("nextPerCommitmentPoint" | publicKey) ::
       ("tlvStream" | FundingLockedTlv.fundingLockedTlvCodec)).as[FundingLocked]
 
+  private val scriptSigOptCodec: Codec[Option[ByteVector]] = lengthDelimited(bytes).xmap[Option[ByteVector]](
+    b => if (b.isEmpty) None else Some(b),
+    b => b.getOrElse(ByteVector.empty)
+  )
+
+  val txAddInputCodec: Codec[TxAddInput] = (
+    ("channelId" | bytes32) ::
+      ("serialId" | uint64) ::
+      ("previousTx" | lengthDelimited(txCodec)) ::
+      ("previousTxOutput" | uint32) ::
+      ("sequence" | uint32) ::
+      ("scriptSig" | scriptSigOptCodec) ::
+      ("tlvStream" | TxAddInputTlv.txAddInputTlvCodec)).as[TxAddInput]
+
+  val txAddOutputCodec: Codec[TxAddOutput] = (
+    ("channelId" | bytes32) ::
+      ("serialId" | uint64) ::
+      ("amount" | satoshi) ::
+      ("pubkeyScript" | lengthDelimited(bytes)) ::
```

Comment:
Bitcoin has notoriously badly named these transaction fields :)
You're right that it named this field `scriptPubKey` in `vout`, so I'll do the same.
[I'm trying to get that field removed from the message anyway](https://github.com/niftynei/lightning-rfc/pull/3#discussion_r847351069), AFAIU it's only needed if we want to support segwit embedded inside P2SH, but I think we should only support segwit inputs.

Reply:
Done in https://github.com/ACINQ/eclair/pull/2231/commits/7b7f24e4e44cda730345284478524da554683537, I'd like to stick to bitcoin terminology instead of just `script`, otherwise you need to think more before understanding which kind of script this is